menpofit/sdm/fitter.py

```diff
@@ -308,14 +336,36 @@ def _train_batch(self, image_batch, increment=False, group=None,
                     scaled_images, scaled_shapes, current_shapes,
                     prefix=scale_prefix, verbose=verbose)
 
-            # Scale current shapes to next resolution, don't bother
-            # scaling final level
-            if j != (self.n_scales - 1):
-                transform = Scale(self.scales[j + 1] / self.scales[j],
-                                  n_dims=2)
-                for image_shapes in current_shapes:
-                    for k, shape in enumerate(image_shapes):
-                        image_shapes[k] = transform.apply(shape)
+            # Scale the current shape estimations for the next level. This
+            # doesn't have to be done for the last scale. The only thing we need
+            # to do at the last scale is to remove any attached landmarks from
+            # the training images.
+            if j < (self.n_scales - 1):
+                if self.holistic_features[j + 1] != self.holistic_features[j]:
+                    # Features will be extracted, thus attach current_shapes on
+                    # the training images (image_batch)
+                    for jj, image_shapes in enumerate(current_shapes):
+                        for k, shape in enumerate(image_shapes):
+                            c_key = '__sdm_current_shape_{}'.format(k)
+                            image_batch[jj].landmarks[c_key] = \
+                                scale_transforms[jj].apply(shape)
+                else:
+                    # Features won't be extracted;. the same feature_images will
+                    # be used for the next scale, thus attach current_shapes on
+                    # them.
+                    for jj, image_shapes in enumerate(current_shapes):
+                        for k, shape in enumerate(image_shapes):
+                            c_key = '__sdm_current_shape_{}'.format(k)
+                            feature_images[jj].landmarks[c_key] = \
+                                scale_transforms[jj].apply(shape)
+            else:
+                # Check if original training image (image_batch) got some current
+                # shape estimations attached. If yes, delete them.
+                if '__sdm_current_shape_0' in image_batch[0].landmarks.keys():
```

Comment:
Here you should be able to just do `if '__sdm_current_shape_0' in image_batch[0].landmarks`